File: state-ui.js
var globalState = {}

class StateUI {
    constructor(element) {
        var element = document.getElementsByTagName(element) 

        globalState[element]
    }

    updateHTML(html) {
        
    }

    updateValue(value) {

    }
}

window.addEventListener('message', function(event) {
    if (globalState[event.data.type] !== null) {
        let ui = globalState[event.data.type].StateUI()
        ui.updateHTML
    } else if (event.data.internal == "use_state") {
        globalState[event.data.element].StateUI = new StateUI(event.data.element)
    }
})

/*

let list = new StateUI("#het")

list.updateValue()

*/

File: state-ui.lua
local state = {}

function use_state(element)
    SendNUIMessage({
        internal = "use_state",
        element = element
    })
    return {
        function() 
            return state[element]
        end,
        function(type, value)
            state[element][value]
            SendNUIMessage({
                type = element,
                state = state[element]
            })
        end
    }
end

local [get_tweets, set_tweets] = use_state("#tweets")

function update_tweet(message, tweet)
    set_tweets("value", "my tweet")
end

function update_tweets(message, tweet)
    set_tweets("html", [[
        <ul>
            <span id="message_738213">]] .. message .. [[</span>
        </ul>
    ]])
end